### PR TITLE
Add and modify module files for AWS

### LINKED ISSUE #936: add module files to compile on AWS
add module files to compile on AWS, Azure and Google CSP.

### PR DESCRIPTION
**Pull request by weihuang-jedi**
Add and modify module files for AWS, as in issue: #936

## TESTS CONDUCTED: 

- [x] Compile branch (2e427c5) on Hera, Orion, Hercules, Jet and WCOSS2 using Intel.
- [x] Compile branch on AWS using Intel.

## ISSUE: 
Fixes #936

### CODE REVIEW

File: build_all.sh [hunk 1]
```diff
@@ -27,17 +29,16 @@ if [[ "$target" == "linux.*" || "$target" == "macosx.*" ]]; then
   set -x
 else
   set +x
-  source "${DIR_ROOT}/sorc/machine-setup.sh"
   if [[ "${target}" == "noaacloud" ]]; then
```
Comment (GeorgeGayno-NOAA) [inline build_all.sh]: Why can't these module use/loads be placed in the "build.noaacloud.intel.lua"? For example, see "build.hercules.intel.lua".

**Comment by weihuang-jedi** [comment]
Let me check on this.

File: build_all.sh (hunk 1 above)
Comment (weihuang-jedi) [reply build_all.sh]: We can restructure "build_all.sh" to make it more like others.
But have problem to load build.noaacloud.intel (the previous one).

So we have to split build.noaacloud.intel.lua into two parts,
the first part kept this name, the second part named common4noaacloud.lua.

Please take a look and see if you like the second approach.

I see other packages have similar approach.

Comment (GeorgeGayno-NOAA) [reply build_all.sh]: Changes at 90e210c look good.

**Comment by weihuang-jedi** [comment]
thank you

File: build_all.sh
Comment (GeorgeGayno-NOAA) [inline build_all.sh:1]: Is the PW_CSP variable used for compiling on the cloud? I don't see where it is used.

Comment (GeorgeGayno-NOAA) [reply build_all.sh:1]: Thanks for removing this unused variable.

**Comment by weihuang-jedi** [comment]
PW_CSP is not needed in build_all.sh.

I have removed it, pushed the change in, and test compiled on hear, and aws.

**Comment by GeorgeGayno-NOAA** [comment]
Branch at 2e427c5 was successfully compiled using Intel on Hercules, WCOSS2, Orion, Jet and Hera.

**Comment by GeorgeGayno-NOAA** [comment]
@weihuang-jedi - Looks good. Any other changes? Otherwise, I will merge.

**Comment by weihuang-jedi** [comment]
No more changes. Thank you.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

File: build_all.sh
Comment (GeorgeGayno-NOAA) [reply build_all.sh:1]: I now see that PW_CSP is used by ./sorc/machine_setup.sh. Does it need to be retained? I just merged this PR. Should I back out the merge?

**Comment by weihuang-jedi** [comment]
It seems OK for now.
Without PW_CSP in build_all.sh, it worked fine on AWS.
I can see:

***@***.***:/contrib/Wei.Huang/dev/ufs-utils-cloud]$ git diff
sorc/machine-setup.sh
diff --git a/sorc/machine-setup.sh b/sorc/machine-setup.sh
index 9d72676..62db25a 100644
--- a/sorc/machine-setup.sh
+++ b/sorc/machine-setup.sh
@@ -94,6 +94,7 @@ elif [[ -d /data/prod ]] ; then
     target=s4
     module purge
 elif [[ "$(dnsdomainname)" =~ "pw" ]]; then
+    echo "PW_CSP: ${PW_CSP}"
     if [[ "${PW_CSP}" == "aws" ]]; then # TODO: Add other CSPs here.
        target=noaacloud
         module purge
***@***.***:/contrib/Wei.Huang/dev/ufs-utils-cloud]$ source
sorc/machine-setup.sh
PW_CSP: aws

So ,machine-setup can still catch the env. var. PW_CSP, and I tried compile
on AWS, it works fine.

**Pull request merged**